File: packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase_noop/InAppPurchasePlugin.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.inapppurchase;

import androidx.annotation.NonNull;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.plugin.common.BinaryMessenger;

/** Registers the no-op Android implementation of the in-app purchase channel. */
public class InAppPurchasePlugin implements FlutterPlugin {

  private NoopInAppPurchaseApi api;

  @Override
  public void onAttachedToEngine(@NonNull FlutterPlugin.FlutterPluginBinding binding) {
    setUpMethodChannel(binding.getBinaryMessenger());
  }

  @Override
  public void onDetachedFromEngine(@NonNull FlutterPlugin.FlutterPluginBinding binding) {
    Messages.InAppPurchaseApi.setUp(binding.getBinaryMessenger(), null);
    api = null;
  }

  private void setUpMethodChannel(BinaryMessenger messenger) {
    api = new NoopInAppPurchaseApi();
    Messages.InAppPurchaseApi.setUp(messenger, api);
  }
}

File: packages/in_app_purchase/in_app_purchase_android/android/src/main/kotlin/io/flutter/plugins/inapppurchase/InAppPurchasePlugin.kt
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package io.flutter.plugins.inapppurchase

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.embedding.engine.plugins.activity.ActivityAware

/** Flutter 要求 main 中存在注册入口；实际能力由 SourceSet 在编译时选择。 */
class InAppPurchasePlugin private constructor(
    delegate: InAppPurchasePluginDelegate,
) : FlutterPlugin by delegate, ActivityAware by delegate {
    constructor() : this(InAppPurchasePluginDelegate())
}
